petab_select/candidate_space.py

```diff
@@ -556,21 +1251,48 @@ def setup_after_model_space_search(self):
         pass
 
 
-class LateralCandidateSpace(ForwardCandidateSpace):
+class LateralCandidateSpace(CandidateSpace):
     """Find models with the same number of estimated parameters."""
 
     method = Method.LATERAL
 
     def __init__(
         self,
         *args,
-        predecessor_model: Union[Model, str],
+        predecessor_model: Union[Model, None],
+        max_number_of_steps: int = 0,
         **kwargs,
     ):
-        super().__init__(*args, predecessor_model=predecessor_model, **kwargs)
+        """
+        Additional args:
+            max_number_of_steps:
+                Maximal allowed number of swap moves. If 0 then there is no maximum.
+        """
+        super().__init__(
+            *args,
+            predecessor_model=predecessor_model,
+            **kwargs,
+        )
+        self.max_number_of_steps = max_number_of_steps
 
     def is_plausible(self, model: Model) -> bool:
+        if self.predecessor_model is None:
+            raise ValueError(
+                f"The predecessor_model is still None. Provide an appropriate predecessor_model"
+            )
+
         distances = self.distances_in_estimated_parameters(model)
+
+        # If max_number_of_steps is non-zero and the number of steps made is
+        # larger then move is not plausible.
+        if (
+            self.max_number_of_steps
+            and distances['l1'] > 2 * self.max_number_of_steps
+        ):
+            raise StopIteration(
+                f"Maximal number of steps for method {self.method} exceeded. Stop sending candidate models."
+            )
```

Comment:
Rewrite with a new method and add it to `CandidateSpace.consider`, such that this `raise StopIteration` can be replaced with a method that results in `CandidateSpace.consider` returning `False` -> this should result in the model subspace stopping its search [1], similar to the current `try-except` blocks in this PR.

[1] https://github.com/PEtab-dev/petab_select/blob/58c5addabb0477f21d54e12c2af3abcac8a46cd7/petab_select/model_subspace.py#L370-L371

Reply:
So I should rewrite consider to return False to continue_sending in case a model is not plausible with the reason of lines 1288-1291? So this would replace the try-except blocks in model_subspace.py that catch StopIteration? Yep, sounds good. I'll think how to replace it.